From 01ca9fdcd67a393d11cbfb1453f4189c379f6caa Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Tue, 1 Dec 2015 16:25:37 +0100
Subject: [PATCH 2/3] Minimize device tree for the S2

The device tree for S2 is minimized. That means components like USDHC2
and GPMI are commented out. The ldo bypass is disabled and the gpio led
on the score is added.

Ticket #805
---
 arch/arm/boot/dts/imx6sx-s2.dts | 158 ++++++++++++++++++----------------------
 1 file changed, 71 insertions(+), 87 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-s2.dts b/arch/arm/boot/dts/imx6sx-s2.dts
index 38f2a55..5216720 100644
--- a/arch/arm/boot/dts/imx6sx-s2.dts
+++ b/arch/arm/boot/dts/imx6sx-s2.dts
@@ -10,20 +10,28 @@
 /dts-v1/;
 
 #include <dt-bindings/input/input.h>
-#include "imx6sx.dtsi"
+#include "imx6sx-score.dtsi"
 
 / {
 	model = "sitec systems S2";
-	compatible = "fsl,imx6sx-score", "fsl,imx6sx";
+	compatible = "fsl,imx6sx-s2", "fsl,imx6sx";
 
 	chosen {
 		stdout-path = &uart1;
 	};
 
 	memory {
-		reg = <0x80000000 0x40000000>;
+		reg = <0x80000000 0x20000000>;
 	};
 
+    leds {
+        compatible = "gpio-leds";
+        alive {
+            label = "Alive";
+            gpios = <&gpio3 19 GPIO_ACTIVE_HIGH>;
+            linux,default-trigger = "heartbeat";
+        };
+    };
 
 	regulators {
 		compatible = "simple-bus";
@@ -59,22 +67,13 @@
 		396000	1175000
 		198000	1175000
 	>;
-	arm-supply = <&sw1a_reg>;
-	soc-supply = <&sw1a_reg>;
-	fsl,arm-soc-shared = <1>;
+	arm-supply = <&reg_arm>;
+	soc-supply = <&reg_soc>;
 };
 
 &gpc {
 	/* use ldo-bypass, u-boot will check it and configure */
-	fsl,ldo-bypass = <1>;
-};
-
-&flexcan2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_flexcan2>;
-	trx-stby-gpio = <&gpio1 02 GPIO_ACTIVE_LOW>;
-	trx_wakeup_gpio = <&gpio1 03 GPIO_ACTIVE_HIGH>;
-	status = "okay";
+	fsl,ldo-bypass = <0>;
 };
 
 &uart1 {
@@ -96,6 +95,14 @@
 	status = "okay";
 };
 
+&flexcan2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_flexcan2>;
+	trx-stby-gpio = <&gpio1 02 GPIO_ACTIVE_LOW>;
+	trx_wakeup_gpio = <&gpio1 03 GPIO_ACTIVE_HIGH>;
+	status = "okay";
+};
+
 &usbotg1 {
 	dr_mode = "host";
 	status = "okay";
@@ -106,25 +113,22 @@
 	status = "okay";
 };
 
-&usdhc2 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_usdhc2>;
-	bus-width = <4>;
-	non-removable;
-	no-1-8-v;
-	keep-power-in-suspend;
-	enable-sdio-wakeup;
-	status = "okay";
-};
+/*&usdhc2 {*/
+	/*pinctrl-names = "default";*/
+	/*pinctrl-0 = <&pinctrl_usdhc2>;*/
+	/*bus-width = <4>;*/
+	/*non-removable;*/
+	/*no-1-8-v;*/
+	/*keep-power-in-suspend;*/
+	/*enable-sdio-wakeup;*/
+	/*status = "okay";*/
+/*};*/
 
 &usdhc3 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc3>;
-	pinctrl-1 = <&pinctrl_usdhc3_100mhz>;
-	pinctrl-2 = <&pinctrl_usdhc3_200mhz>;
 	bus-width = <4>;
-	keep-power-in-suspend;
-	enable-sdio-wakeup;
+    broken-cd;
 	status = "okay";
 };
 
@@ -136,6 +140,7 @@
 
 	led1: lp5562@30 {
 		compatible = "ti,lp5562";
+        label = "LED1";
 		reg = <0x30>;
 		clock-mode = /bits/8 <1>;
 
@@ -158,6 +163,7 @@
 	};
 	led2: lp5562@31 {
 		compatible = "ti,lp5562";
+        label = "LED2";
 		reg = <0x31>;
 		clock-mode = /bits/8 <1>;
 
@@ -282,12 +288,12 @@
 	status = "okay";
 };
 
-&gpmi {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_gpmi_nand_1>;
-	status = "okay";
-	nand-on-flash-bbt;
-};
+/*&gpmi {*/
+	/*pinctrl-names = "default";*/
+	/*pinctrl-0 = <&pinctrl_gpmi_nand_1>;*/
+	/*status = "okay";*/
+	/*nand-on-flash-bbt;*/
+/*};*/
 
 &iomuxc {
 	pinctrl-names = "default";
@@ -330,25 +336,25 @@
 			>;
 		};
 
-		pinctrl_gpmi_nand_1: gpmi-nand-1 {
-			fsl,pins = <
-				MX6SX_PAD_NAND_CLE__RAWNAND_CLE         0xb0b1
-				MX6SX_PAD_NAND_ALE__RAWNAND_ALE         0xb0b1
-				MX6SX_PAD_NAND_WP_B__RAWNAND_WP_B       0xb0b1
-				MX6SX_PAD_NAND_READY_B__RAWNAND_READY_B 0xb000
-				MX6SX_PAD_NAND_CE0_B__RAWNAND_CE0_B     0xb0b1
-				MX6SX_PAD_NAND_RE_B__RAWNAND_RE_B       0xb0b1
-				MX6SX_PAD_NAND_WE_B__RAWNAND_WE_B       0xb0b1
-				MX6SX_PAD_NAND_DATA00__RAWNAND_DATA00   0xb0b1
-				MX6SX_PAD_NAND_DATA01__RAWNAND_DATA01   0xb0b1
-				MX6SX_PAD_NAND_DATA02__RAWNAND_DATA02   0xb0b1
-				MX6SX_PAD_NAND_DATA03__RAWNAND_DATA03   0xb0b1
-				MX6SX_PAD_NAND_DATA04__RAWNAND_DATA04   0xb0b1
-				MX6SX_PAD_NAND_DATA05__RAWNAND_DATA05   0xb0b1
-				MX6SX_PAD_NAND_DATA06__RAWNAND_DATA06   0xb0b1
-				MX6SX_PAD_NAND_DATA07__RAWNAND_DATA07   0xb0b1
-			>;
-		};
+		/*pinctrl_gpmi_nand_1: gpmi-nand-1 {*/
+			/*fsl,pins = <*/
+				/*MX6SX_PAD_NAND_CLE__RAWNAND_CLE         0xb0b1*/
+				/*MX6SX_PAD_NAND_ALE__RAWNAND_ALE         0xb0b1*/
+				/*MX6SX_PAD_NAND_WP_B__RAWNAND_WP_B       0xb0b1*/
+				/*MX6SX_PAD_NAND_READY_B__RAWNAND_READY_B 0xb000*/
+				/*MX6SX_PAD_NAND_CE0_B__RAWNAND_CE0_B     0xb0b1*/
+				/*MX6SX_PAD_NAND_RE_B__RAWNAND_RE_B       0xb0b1*/
+				/*MX6SX_PAD_NAND_WE_B__RAWNAND_WE_B       0xb0b1*/
+				/*MX6SX_PAD_NAND_DATA00__RAWNAND_DATA00   0xb0b1*/
+				/*MX6SX_PAD_NAND_DATA01__RAWNAND_DATA01   0xb0b1*/
+				/*MX6SX_PAD_NAND_DATA02__RAWNAND_DATA02   0xb0b1*/
+				/*MX6SX_PAD_NAND_DATA03__RAWNAND_DATA03   0xb0b1*/
+				/*MX6SX_PAD_NAND_DATA04__RAWNAND_DATA04   0xb0b1*/
+				/*MX6SX_PAD_NAND_DATA05__RAWNAND_DATA05   0xb0b1*/
+				/*MX6SX_PAD_NAND_DATA06__RAWNAND_DATA06   0xb0b1*/
+				/*MX6SX_PAD_NAND_DATA07__RAWNAND_DATA07   0xb0b1*/
+			/*>;*/
+		/*};*/
 
 		pinctrl_gsm_state: gsm-state {
 			fsl,pins = <
@@ -426,16 +432,16 @@
 			>;
 		};
 
-		pinctrl_usdhc2: usdhc2grp {
-			fsl,pins = <
-				MX6SX_PAD_SD2_CMD__USDHC2_CMD		0x17059
-				MX6SX_PAD_SD2_CLK__USDHC2_CLK		0x10059
-				MX6SX_PAD_SD2_DATA0__USDHC2_DATA0	0x17059
-				MX6SX_PAD_SD2_DATA1__USDHC2_DATA1	0x17059
-				MX6SX_PAD_SD2_DATA2__USDHC2_DATA2	0x17059
-				MX6SX_PAD_SD2_DATA3__USDHC2_DATA3	0x17059
-			>;
-		};
+		/*pinctrl_usdhc2: usdhc2grp {*/
+			/*fsl,pins = <*/
+				/*MX6SX_PAD_SD2_CMD__USDHC2_CMD		0x17059*/
+				/*MX6SX_PAD_SD2_CLK__USDHC2_CLK		0x10059*/
+				/*MX6SX_PAD_SD2_DATA0__USDHC2_DATA0	0x17059*/
+				/*MX6SX_PAD_SD2_DATA1__USDHC2_DATA1	0x17059*/
+				/*MX6SX_PAD_SD2_DATA2__USDHC2_DATA2	0x17059*/
+				/*MX6SX_PAD_SD2_DATA3__USDHC2_DATA3	0x17059*/
+			/*>;*/
+		/*};*/
 
 		pinctrl_usdhc3: usdhc3grp {
 			fsl,pins = <
@@ -447,27 +453,5 @@
 				MX6SX_PAD_SD3_DATA3__USDHC3_DATA3	0x17069
 			>;
 		};
-
-		pinctrl_usdhc3_100mhz: usdhc3grp-100mhz {
-			fsl,pins = <
-				MX6SX_PAD_SD3_CMD__USDHC3_CMD		0x170b9
-				MX6SX_PAD_SD3_CLK__USDHC3_CLK		0x100b9
-				MX6SX_PAD_SD3_DATA0__USDHC3_DATA0	0x170b9
-				MX6SX_PAD_SD3_DATA1__USDHC3_DATA1	0x170b9
-				MX6SX_PAD_SD3_DATA2__USDHC3_DATA2	0x170b9
-				MX6SX_PAD_SD3_DATA3__USDHC3_DATA3	0x170b9
-			>;
-		};
-
-		pinctrl_usdhc3_200mhz: usdhc3grp-200mhz {
-			fsl,pins = <
-				MX6SX_PAD_SD3_CMD__USDHC3_CMD		0x170f9
-				MX6SX_PAD_SD3_CLK__USDHC3_CLK		0x100f9
-				MX6SX_PAD_SD3_DATA0__USDHC3_DATA0	0x170f9
-				MX6SX_PAD_SD3_DATA1__USDHC3_DATA1	0x170f9
-				MX6SX_PAD_SD3_DATA2__USDHC3_DATA2	0x170f9
-				MX6SX_PAD_SD3_DATA3__USDHC3_DATA3	0x170f9
-			>;
-		};
 	};
 };
-- 
1.9.1

